### PR TITLE
Update permission ADR based on prototype.

### PR DESCRIPTION
**Pull request by TingluoHuang**
Updates were found during:
- https://github.com/actions/actions-runner-controller/pull/2276
- https://github.com/actions/actions-runner-controller/pull/2363
- https://github.com/actions/actions-runner-controller/pull/2374

### CODE REVIEW

File: adrs/2023-02-10-limit-manager-role-permission.md [hunk 1]
```diff
@@ -66,18 +66,54 @@ To help these customers and improve security for `actions-runner-controller` in
 - List/Watch on `RoleBindings`
 - List/Watch on `ServiceAccounts`
 
-> We will change the default cache-based client to bypass cache on reading `Secrets`, so we can eliminate the need for `List` and `Watch` `Secrets` permission in cluster scope.
+> We will change the default cache-based client to bypass cache on reading `Secrets` and `ConfigMaps`(ConfigMap is used when you configure `githubServerTLS`), so we can eliminate the need for `List` and `Watch` `Secrets` permission in cluster scope.
 
-Introduce a new `Role` per `AutoScalingRunnerSet` installation and `RoleBinding` the `Role` with the controller's service account in the namespace that each `AutoScalingRunnerSet` deployed with the following permission.
+Introduce a new `Role` for the controller and `RoleBinding` the `Role` with the controller's service account in the namespace the controller is deployed, this role will grant the controller service account required permission to work with `AutoScalingListeners` in the controller namespace.
```
Comment (nikola-jokic) [inline adrs/2023-02-10-limit-manager-role-permission.md]:
```suggestion
Introduce a new `Role` for the controller and `RoleBinding` the `Role` with the controller's `ServiceAccount` in the namespace the controller is deployed. This role will grant the controller's service account required permission to work with `AutoScalingListeners` in the controller namespace.
```
nit

File: adrs/2023-02-10-limit-manager-role-permission.md [hunk 2]
```diff
@@ -66,18 +66,54 @@ To help these customers and improve security for `actions-runner-controller` in
...
+The `Role` and `RoleBinding` creation will happen during the `helm install demo oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set-controller`
+
+During `helm install demo oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set-controller`, we will store the controller's service account info as labels on the controller `Deployment`.
+Ex:
+```yaml
+    actions.github.com/controller-service-account-namespace: {{ .Release.Namespace }}
+    actions.github.com/controller-service-account-name: {{ include "gha-runner-scale-set-controller.serviceAccountName" . }}
+```
+
+Introduce a new `Role` per `AutoScalingRunnerSet` installation and `RoleBinding` the `Role` with the controller's service account in the namespace that each `AutoScalingRunnerSet` deployed with the following permission.
```
Comment (nikola-jokic) [inline adrs/2023-02-10-limit-manager-role-permission.md]:
```suggestion
Introduce a new `Role` per `AutoScalingRunnerSet` installation and `RoleBinding` the `Role` with the controller's `ServiceAccount` in the namespace that each `AutoScalingRunnerSet` deployed with the following permission.
```
nit: if you don't agree with the previous suggestion, feel free to ignore this one as well :relaxed:

File: adrs/2023-02-10-limit-manager-role-permission.md [hunk 3]
```diff
@@ -86,10 +122,12 @@ In this mode, the `actions-runner-controller` will only be able to watch the `Au
 
 If you want to deploy multiple `AutoScalingRunnerSet` into different namespaces, you will need to install `actions-runner-controller` in this mode multiple times as well and have each installation watch the namespace you want to deploy an `AutoScalingRunnerSet`
 
-You will install `actions-runner-controller` with something like `helm install arc --set watchSingleNamespace=TestNamespace oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set-controller` (the `TestNamespace` namespace needs to be created first).
+You will install `actions-runner-controller` with something like `helm install arc --namespace arc-system --set watchSingleNamespace=TestNamespace oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set-controller` (the `TestNamespace` namespace needs to be created first).
```
Comment (nikola-jokic) [inline adrs/2023-02-10-limit-manager-role-permission.md]:
```suggestion
You will install `actions-runner-controller` with something like `helm install arc --namespace arc-system --set watchSingleNamespace=test-namespace oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set-controller` (the `test-namespace` namespace needs to be created first).
```
Namespace must be `[a-z0-9]([-a-z0-9]*[a-z0-9])?`

**Review by rentziass — DISMISSED** [review]
Looking good 🙌

File: adrs/2023-02-10-limit-manager-role-permission.md [hunk 4]
```diff
@@ -86,10 +122,12 @@ In this mode, the `actions-runner-controller` will only be able to watch the `Au
...
+You will install `actions-runner-controller` with something like `helm install arc --namespace arc-system --set watchSingleNamespace=test-namespace oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set-controller` (the `test-namespace` namespace needs to be created first).
 
 You will deploy the `AutoScalingRunnerSet` with something like `helm install demo --namespace TestNamespace oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set`
 
-In this mode, you will end up with a manager `Role` that has all Get/List/Create/Delete/Update/Patch/Watch permissions on resources we need, and a `RoleBinding` to bind the `Role` with the controller `ServiceAccount` in the watched single namespace, ex: `TestNamespace` in the above example.
+In this mode, you will end up with a manager `Role` that has all Get/List/Create/Delete/Update/Patch/Watch permissions on resources we need, and a `RoleBinding` to bind the `Role` with the controller `ServiceAccount` in the watched single namespace and the controller namespace, ex: `TestNamespace` and `arc-system` in the above example.
```
Comment (nikola-jokic) [inline adrs/2023-02-10-limit-manager-role-permission.md]:
```suggestion
In this mode, you will end up with a manager `Role` that has all Get/List/Create/Delete/Update/Patch/Watch permissions on resources we need, and a `RoleBinding` to bind the `Role` with the controller `ServiceAccount` in the watched single namespace and the controller namespace, ex: `test-namespace` and `arc-system` in the above example.
```

**Review by nikola-jokic — APPROVED** [review]
LGTM

**Pull request merged**